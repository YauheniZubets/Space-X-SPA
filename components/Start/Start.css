main {
    /* height: 500px; */
    display: flex;
    justify-content: center;
    align-items: center;
}
.Start {
    width: 500px;
    height: 500px;
    background-image: url('/public/logo.svg');
}
.Start-Earth{
    width: 90%;
    height: inherit;
    display: flex;
    justify-content: center;
    align-items: center;
}
.Start-Earth img {
    /* position: relative;
    left: 50%; */
}
.Start-video-display-none {
    display: none;
}