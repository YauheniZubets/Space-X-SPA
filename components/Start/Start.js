import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Preloadcircle from '../Preloadcircle/Preloadcircle';

import './Start.css';

class Start extends React.PureComponent {

    static propTypes = {
    }

    state={
        opacity : 0, //прозрачность
        startLogo: true, //показать логотип space-x
        mainPreloader: false // проверяем загрузилось ли видео, если загружено прелоадер удаляется

    }

    changeOpacity = () => { //прозрачность div
        let timer=setTimeout(()=>{
            this.setState({opacity:1});
            (this.state.opacity>=1 && clearTimeout(timer));
            let timeOutdiv=setTimeout(()=>this.setState({startLogo: false}), 2500);
            (!this.state.startLogo && clearTimeout(timeOutdiv));//убираем див
        }, 1000);
    }

    cbHidePreloader = () => {
        console.log('hide');
        this.setState({mainPreloader: true});
    }

    componentDidMount(){
        this.changeOpacity();
    }

    componentWillReceiveProps() { // сброс состояния прелоадера для image
        this.setState({isVideoLoaded: false})
    }

    render () {
        return (
            <main>
                {
                    (this.state.startLogo) && 
                    <div className='Start'>
                        <div style={{opacity: this.state.opacity, backgroundColor: 'black', height: 'inherit', transition: '2.5s linear', transitionProperty: 'opacity'}}></div>
                    </div>
                }
                {
                    (!this.state.startLogo) &&
                    <div className='Start-Earth'>
                        <Preloadcircle hide={this.state.mainPreloader} />
                        <img 
                            src='/public/earth.gif'
                            onLoad={this.cbHidePreloader}
                            className={this.state.mainPreloader ? '' : 'Start-video-display-none'}
                        />
                    </div>
                }
            </main>
            
            
        );
    }
};

export default Start;