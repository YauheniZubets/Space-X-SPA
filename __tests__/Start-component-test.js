import React from 'react';
import renderer from 'react-test-renderer';

import Start from '../components/Start/Start';

test('работа StartComponent', () => {

  const start = renderer.create(
    <Start rockets={[
      {name: "Falcon 1"},
      {name: "Falcon 9"},
      {name: "Falcon Heavy"},
      {name: "Starship"}
    ]} userInfo='Имя пользователя для теста' />
  );

  let componentTree=start.toJSON();
  
  expect(componentTree).toMatchSnapshot();
  
  expect(componentTree).toMatchSnapshot();
    
});
