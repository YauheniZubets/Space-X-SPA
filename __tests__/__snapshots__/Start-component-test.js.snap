// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа StartComponent 1`] = `
<main>
  <div
    className="Start"
  >
    <div
      style={
        Object {
          "backgroundColor": "black",
          "height": "inherit",
          "opacity": 0,
          "transition": "2.5s linear",
          "transitionProperty": "opacity",
        }
      }
    />
  </div>
</main>
`;

exports[`работа StartComponent 2`] = `
<main>
  <div
    className="Start"
  >
    <div
      style={
        Object {
          "backgroundColor": "black",
          "height": "inherit",
          "opacity": 0,
          "transition": "2.5s linear",
          "transitionProperty": "opacity",
        }
      }
    />
  </div>
</main>
`;
